types/types.go

```diff
@@ -485,7 +495,9 @@ type SystemContext struct {
 	DockerPerHostCertDirPath string
 	// Allow contacting docker registries over HTTP, or HTTPS with failed TLS verification. Note that this does not affect other TLS connections.
 	DockerInsecureSkipTLSVerify OptionalBool
-	// if nil, the library tries to parse ~/.docker/config.json to retrieve credentials
+	// if nil, the library tries to parse ~/.docker/config.json to retrieve credentials.
+	//
+	// Deprecated: Use the GetAuth property and github.com/containers/image/credentials/single's AuthStore instead.
```

Comment:
It’s not obviously attractive to force the users to create extra objects when they could just set a clearly-named field. Sure, the consumer of the data might need to define what happens if multiple fields are set, or refuse such configurations, but that seems worth it.

Plumbing together creation of a `single.Credential`+`single.AuthStore`, and _somehow_ making an `AuthGetter` out of that, seems way too much code compared to a `DockerAuthConfig: &types.DockerAuthConfig{…}`

Reply:
> Plumbing together creation of a `single.Credential`+`single.AuthStore`, and somehow making an `AuthGetter` out of that...

```console
$ git describe
v1.3-49-g44801dd
$ cat test.go
package main

import (
	"fmt"

	"github.com/containers/image/types"
	"github.com/containers/image/credentials/single"
	"github.com/docker/docker-credential-helpers/credentials"
)

func main() {
	authStore := single.AuthStore(credentials.Credentials{
		ServerURL: "example.org",
		Username: "alice",
		Secret: "secret",
	})
	sys := &types.SystemContext{
		GetAuth: authStore.Get,
	}
	username, secret, err := sys.GetAuth("example.org")
	fmt.Println(username)
	fmt.Println(secret)
	fmt.Println(err)
}
$ go run test.go 
alice
secret
<nil>
```

That doesn't seem too complicated to me, and it saves you from having to explain how `DockerAuthConfig` interacts with a `GetAuth` or other `SystemContext` properties.